### ISSUE TITLE
Veegetable Stew naming issue

### ISSUE THREAD

**Issue by Burgersim** [anegostudios/VintageStory-Issues#2690]
### Game Version

v1.18.1

### Platform

Windows

### Modded

Modded

### SP/MP

Multiplayer

### Description

Hi there, I hope this is not the wrong place to ask and if this is not a game issue then I am sorry, but I get naming errors with vegetable stews if I use vegetables as main ingredients, mushrooms, turnips, you name it. We do have mods installed, but also if I only use vanilla ingredients it still doesn't show a "proper" name, but something like in the screenshot (e.g. turnip.meal-ingredient-secondary-mushroom-deer-ear with rhubarb garnish in pot or only using deer-ear mushrooms gives me meal-ingredient-stew-primary-mushroom-stew)

### How to reproduce

1. Make Vegetable Stew

### Screenshots

![2023-05-01_18-53-36](https://user-images.githubusercontent.com/26793150/235492319-c4f75b24-52d7-4bc1-8a24-9ca509c5c23b.png)

### Logs

```Log```

**Comment by Craluminum2413** [comment]
Does it happen without any mods installed?

**Comment by Burgersim** [comment]
Yes it does happen with unmodded game as well, e.g. when I use Almond Mushrooms as main ingredients and then Oyster as an addon

**Comment by noelle-lavenza** [comment]
Can you upload your client-debug log?

**Comment by noelle-lavenza** [comment]
Think I found the root cause, no need. `VanillaCookingRecipeNames.getMainIngredientName` uses `Lang.Get` which doesn't resolve wildcards, causing issues with any ingredient lang entry that includes wildcards, like the one for deer ear. The solution is to change it to `Lang.GetMatching`.

Why did this work in 1.17? Because of a now-fixed (via a workaround) bug in `Lang.HasTranslation`, where it wouldn't check wildcards with the domain prefix. This would make it return false and trigger the fallback check:

```csharp
            langcode = $"meal-ingredient-{code}-{t}-{itemstack.Collectible.FirstCodePart()}";
            return Lang.Get(langcode);
```

which would lead to, for example, `"meal-ingredient-stew-primary-mushroom": "mushroom"` for Deer Ear mushrooms, which doesn't have a wildcard and thus resolves properly.

Also as a side-note, the workaround in HasTranslation could probably be replaced with just using `validKey` instead.
```diff
-                if (!key.Contains(":")) key = "game:" + key;
-                bool result = wildcardCache.Any(pair => key.StartsWithFast(pair.Key));
+                bool result = wildcardCache.Any(pair => validKey.StartsWithFast(pair.Key));
                 if (!result) result = regexCache.Values.Any(pair => pair.Key.IsMatch(validKey));
                 if (!result && logErrors && !key.Contains("desc-") && notFound.Add(key)) logger.VerboseDebug("Lang key not found: " + key.Replace("{", "{{").Replace("}", "}}"));
```

**Comment by radfast** [comment]
Thank you @Burgersim, thank you @noelle-lavenza .   This issue will be fixed in 1.18.4 coming out later today
